sources/web/datalab/polymer/components/datalab-data/datalab-data.ts

```diff
@@ -0,0 +1,210 @@
+ * Contains a form for initial user input, and a list of datasets.
+ */
+class DataElement extends Polymer.Element {
+
+  public projectValue : string;
+  public collectionsFilterValue : string;
+  public datasetsFilterValue : string;
+
+  private _collectionsList : Array<Collection>;
+  private _datasetsList : Array<Dataset>;
+
+  static get is() { return "datalab-data"; }
+
+  static get properties() {
+    return {
+      projectValue: {
+        type: String,
+        value: 'fake-project',
+      },
+      collectionsFilterValue: {
+        type: String,
+        value: 'no-filter',
+      },
+      datasetsFilterValue: {
+        type: String,
+        value: 'no-filter',
+      },
+    };
+  }
+
+  ready() {
+    super.ready();
+
+    this._collectionsList = [];
+    const collectionsElement = this.shadowRoot.querySelector('#collections')
+    if (collectionsElement) {
+      collectionsElement.addEventListener('itemDoubleClick',
+                                    this._collectionsDoubleClicked.bind(this));
+      collectionsElement.addEventListener('selected-indices-changed',
+                                    this._collectionsSelectionChanged.bind(this));
+    }
+
+    this._datasetsList = [];
+    const datasetsElement = this.shadowRoot.querySelector('#datasets')
+    if (datasetsElement) {
+      datasetsElement.addEventListener('itemDoubleClick',
+                                    this._datasetsDoubleClicked.bind(this));
+      datasetsElement.addEventListener('selected-indices-changed',
+                                    this._datasetsSelectionChanged.bind(this));
+    }
+  }
+
+  _generateFakeCollectionsListForTesting() {
+    const collectionsList = [];
+    const count = Math.floor(Math.random() * 20);
+    for (let i=0; i < count; i++) {
+      collectionsList.push(this._generateFakeCollectionForTesting());
+    }
+    return collectionsList;
+  }
+
+  _generateFakeCollectionForTesting() {
+    const fakeCollection = {
+      name: 'fakeCollection' + Math.floor(Math.random() * 1000000),
+      status: 'fake',
+    };
+    return fakeCollection;
+  }
+
+  _generateFakeDatasetsListForTesting() {
+    const datasetsList = [];
+    const count = Math.floor(Math.random() * 20);
+    for (let i=0; i < count; i++) {
+      datasetsList.push(this._generateFakeDatasetForTesting());
+    }
+    return datasetsList;
+  }
+
+  _generateFakeDatasetForTesting() {
+    const fakeDataset = {
+      name: 'fakeDataset' + Math.floor(Math.random() * 1000000),
+      status: 'fake',
+    };
+    return fakeDataset;
+  }
+
+  /** Reads the user's query values, queries for datasets, and updates the results. */
+  _search() {
+    this._collectionsList = this._generateFakeCollectionsListForTesting();
+    this._renderCollectionsList();
+  }
+
+  /**
+   * Creates a new ItemListRow object for each entry in the collections list, and sends
+   * the created list to the item-list to render.
+   */
+  _renderCollectionsList() {
```

Comment:
This should probably be an observable on `_collectionsList`, instead of calling it manually every time the list changes.